### ISSUE TITLE
devenv warning "is newer than devenv input in devenv.lock" does not disappear after following the warning instructions

### ISSUE THREAD

**Issue by jozuas** [cachix/devenv#1501]
**Describe the bug**

devenv warning "is newer than devenv input in devenv.lock" does not disappear after following the warning instructions

```
juozas@macos> direnv reload                                                                                                                                   
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.

juozas@macos> devenv update                                                                                                                                     
• Updating devenv.lock ...
warning: updating lock file '/Users/juozas/dev/terrapin-app/devenv.lock':
• Updated input 'agenix/darwin':
    'github:lnl7/nix-darwin/6374cd7e50aa057a688142eed2345083047ad884' (2024-09-17)
  → 'github:lnl7/nix-darwin/8c8388ade72e58efdeae71b4cbb79e872c23a56b' (2024-10-03)
• Updated input 'agenix/home-manager':
    'github:nix-community/home-manager/d2493de5cd1da06b6a4c3e97f4e7d5dd791df457' (2024-09-17)
  → 'github:nix-community/home-manager/bc623830e619cef86a2d3750625ffe4e24ea7e64' (2024-10-07)
• Updated input 'agenix/nixpkgs':
    'github:NixOS/nixpkgs/99dc8785f6a0adac95f5e2ab05cc2e1bf666d172' (2024-09-16)
  → 'github:NixOS/nixpkgs/bc947f541ae55e999ffdb4013441347d83b00feb' (2024-10-04)
• Updated input 'devenv':
    'github:cachix/devenv/5cac85ca427b5461a657472ffc4d15d0a61a5ef2?dir=src/modules' (2024-09-18)
  → 'github:cachix/devenv/f1d198c4f56a4d967e7f1c8ba53e210655506b42?dir=src/modules' (2024-10-07)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/7570de7b9b504cfe92025dd1be797bf546f66528' (2024-09-05)
  → 'github:cachix/pre-commit-hooks.nix/1211305a5b237771e13fcca0c51e60ad47326a9a' (2024-10-05)
• Updated input 'pre-commit-hooks/nixpkgs-stable':
    'github:NixOS/nixpkgs/086b448a5d54fd117f4dc2dee55c9f0ff461bdc1' (2024-09-16)
  → 'github:NixOS/nixpkgs/ecbc1ca8ffd6aea8372ad16be9ebbb39889e55b6' (2024-10-06)
✔ Updating devenv.lock in 2.6s.
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
juozas@macos> 

juozas@macos> direnv reload                                                                                                                                   
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
```

**To reproduce**

I'm not sure how to reproduce this given that an older version of devenv has to be used to create a project, then devenv has to be updated to see this. This problem does not seem to be isolated to my machine as I've seen the same happen to my colleagues.

**Version**
 
```
[130] juozas@macos> devenv version                                                                                                                              
devenv 1.3.0 (aarch64-darwin)
```

**Comment by domenkozar** [comment]
Can you show the contents of `.envrc`?

**Comment by jozuas** [comment]
Sure!

```
juozas@macos> cat .envrc                                                                                                                                         
source_url "https://raw.githubusercontent.com/cachix/devenv/d1f7b48e35e6dee421cfd0f51481d17f77586997/direnvrc" "sha256-YBzqskFZxmNb3kYVoKD9ZixoPXJh1C9ZvTLGFRkauZ0="
```

**Comment by domenkozar** [comment]
Can you update it to https://github.com/cachix/devenv/blob/main/devenv/init/.envrc

**Comment by jozuas** [comment]
I updated `.envrc`, but it does not seem like anything changed
<img width="1459" alt="image" src="https://github.com/user-attachments/assets/4d8a28d6-8f0f-42e6-b056-07c4457e7d6c">

Log:
```
[141] juozas@macos> vim .envrc                                                                                                                                   ~/dev/terrapin-app
direnv: error /Users/juozas/dev/terrapin-app/.envrc is blocked. Run `direnv allow` to approve its content
juozas@macos> direnv allow                                                                                                                                       ~/dev/terrapin-app
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
juozas@macos> devenv update                                                                                                                                      ~/dev/terrapin-app
• Updating devenv.lock ...
warning: updating lock file '/Users/juozas/dev/terrapin-app/devenv.lock':
• Updated input 'agenix/darwin':
    'github:lnl7/nix-darwin/6374cd7e50aa057a688142eed2345083047ad884' (2024-09-17)
  → 'github:lnl7/nix-darwin/48b50b3b137be5cfb9f4d006835ce7c3fe558ccc' (2024-10-08)
• Updated input 'agenix/home-manager':
    'github:nix-community/home-manager/d2493de5cd1da06b6a4c3e97f4e7d5dd791df457' (2024-09-17)
  → 'github:nix-community/home-manager/038630363e7de57c36c417fd2f5d7c14773403e4' (2024-10-07)
• Updated input 'agenix/nixpkgs':
    'github:NixOS/nixpkgs/99dc8785f6a0adac95f5e2ab05cc2e1bf666d172' (2024-09-16)
  → 'github:NixOS/nixpkgs/c31898adf5a8ed202ce5bea9f347b1c6871f32d1' (2024-10-06)
• Updated input 'devenv':
    'github:cachix/devenv/5cac85ca427b5461a657472ffc4d15d0a61a5ef2?dir=src/modules' (2024-09-18)
  → 'github:cachix/devenv/e7a0bc559f5a66aaa42f5028db6162c4dd4587eb?dir=src/modules' (2024-10-09)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/7570de7b9b504cfe92025dd1be797bf546f66528' (2024-09-05)
  → 'github:cachix/pre-commit-hooks.nix/1211305a5b237771e13fcca0c51e60ad47326a9a' (2024-10-05)
• Updated input 'pre-commit-hooks/nixpkgs-stable':
    'github:NixOS/nixpkgs/086b448a5d54fd117f4dc2dee55c9f0ff461bdc1' (2024-09-16)
  → 'github:NixOS/nixpkgs/1bfbbbe5bbf888d675397c66bfdb275d0b99361c' (2024-10-07)
✔ Updating devenv.lock in 38.8s.
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
juozas@macos> direnv reload                                                                                                                                      ~/dev/terrapin-app
• Using Cachix: devenv
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
```

**Comment by domenkozar** [comment]
What's in your `devenv.yaml`?

**Comment by jozuas** [comment]
```
juozas@macos> devenv shell
• Building shell ...
• Using Cachix: devenv
✔ Building shell in 0.2s.
• Entering shell
✨ devenv 1.3.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
(devenv) bash-5.2$
```

**Comment by domenkozar** [comment]
Can you share `devenv.nix` and `devenv.lock`?

**Comment by jozuas** [comment]
`devenv.lock` after update:
```json
{
  "nodes": {
    "agenix": {
      "inputs": {
        "darwin": "darwin",
        "home-manager": "home-manager",
        "nixpkgs": "nixpkgs",
        "systems": "systems"
      },
      "locked": {
        "lastModified": 1723293904,
        "owner": "ryantm",
        "repo": "agenix",
        "rev": "f6291c5935fdc4e0bef208cfc0dcab7e3f7a1c41",
        "type": "github"
      },
      "original": {
        "owner": "ryantm",
        "repo": "agenix",
        "rev": "f6291c5935fdc4e0bef208cfc0dcab7e3f7a1c41",
        "type": "github"
      }
    },
    "darwin": {
      "inputs": {
        "nixpkgs": [
          "agenix",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1728769175,
        "owner": "lnl7",
        "repo": "nix-darwin",
        "rev": "fd0e3ed30b75ddf7f3d94829d80a078b413b6244",
        "type": "github"
      },
      "original": {
        "owner": "lnl7",
        "ref": "master",
        "repo": "nix-darwin",
        "type": "github"
      }
    },
    "devenv": {
      "locked": {
        "dir": "src/modules",
        "lastModified": 1728740426,
        "owner": "cachix",
        "repo": "devenv",
        "rev": "ef61728d91ad5eb91f86cdbcc16070602e7afa16",
        "type": "github"
      },
      "original": {
        "dir": "src/modules",
        "owner": "cachix",
        "repo": "devenv",
        "type": "github"
      }
    },
    "flake-compat": {
      "flake": false,
      "locked": {
        "lastModified": 1696426674,
        "owner": "edolstra",
        "repo": "flake-compat",
        "rev": "0f9255e01c2351cc7d116c072cb317785dd33b33",
        "type": "github"
      },
      "original": {
        "owner": "edolstra",
        "repo": "flake-compat",
        "type": "github"
      }
    },
    "gitignore": {
      "inputs": {
        "nixpkgs": [
          "pre-commit-hooks",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1709087332,
        "owner": "hercules-ci",
        "repo": "gitignore.nix",
        "rev": "637db329424fd7e46cf4185293b9cc8c88c95394",
        "type": "github"
      },
      "original": {
        "owner": "hercules-ci",
        "repo": "gitignore.nix",
        "type": "github"
      }
    },
    "home-manager": {
      "inputs": {
        "nixpkgs": [
          "agenix",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1728791962,
        "owner": "nix-community",
        "repo": "home-manager",
        "rev": "64c6325b28ebd708653dd41d88f306023f296184",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "home-manager",
        "type": "github"
      }
    },
    "nixpkgs": {
      "locked": {
        "lastModified": 1728492678,
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "5633bcff0c6162b9e4b5f1264264611e950c8ec7",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nixpkgs-stable": {
      "locked": {
        "lastModified": 1728740863,
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "a3f9ad65a0bf298ed5847629a57808b97e6e8077",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-24.05",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nixpkgs_2": {
      "locked": {
        "lastModified": 1727394691,
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "53488412621165b65bc27627fdb4057d65458d97",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "53488412621165b65bc27627fdb4057d65458d97",
        "type": "github"
      }
    },
    "pre-commit-hooks": {
      "inputs": {
        "flake-compat": "flake-compat",
        "gitignore": "gitignore",
        "nixpkgs": [
          "nixpkgs"
        ],
        "nixpkgs-stable": "nixpkgs-stable"
      },
      "locked": {
        "lastModified": 1728778939,
        "owner": "cachix",
        "repo": "pre-commit-hooks.nix",
        "rev": "ff68f91754be6f3427e4986d7949e6273659be1d",
        "type": "github"
      },
      "original": {
        "owner": "cachix",
        "repo": "pre-commit-hooks.nix",
        "type": "github"
      }
    },
    "root": {
      "inputs": {
        "agenix": "agenix",
        "devenv": "devenv",
        "nixpkgs": "nixpkgs_2",
        "pre-commit-hooks": "pre-commit-hooks"
      }
    },
    "systems": {
      "locked": {
        "lastModified": 1681028828,
        "owner": "nix-systems",
        "repo": "default",
        "rev": "da67096a3b9bf56a91d16901293e51ba5b49a27e",
        "type": "github"
      },
      "original": {
        "owner": "nix-systems",
        "repo": "default",
        "type": "github"
      }
    }
  },
  "root": "root",
  "version": 7
}
```

devenv.nix
```nix
{
  pkgs,
  agenix,
  lib,
  ...
}:

let
  tpDeps = import ./nix/prod/extra-configuration/tp-app-deps.nix { inherit pkgs; };
in
{
  # See full reference at https://devenv.sh/reference/options/

  languages = {
    nix.enable = true;
    shell.enable = true;
    elixir = {
      enable = true;
      package = tpDeps.elixir;
    };
  };

  packages = [
    pkgs.nixfmt-rfc-style

    # At the moment the newest livebook does not work in prod,
    # we backport an older version
    (import ./nix/dev/packages/livebook.nix {
      inherit pkgs;
      elixir = tpDeps.elixir;
      erlang = tpDeps.erlang;
    })

    agenix.packages.${pkgs.stdenv.system}.default
  ] ++ tpDeps.terrapin-app-deps;

  env = {
    ERL_AFLAGS = "-kernel shell_history enabled";
  };

  # Recently there was a change to our env where `devenv up` started running process-manager
  # for managing devenv defined processes. The following line of config is needed to mark
  # PostgreSQL health as "ready". If it is not marked as ready, process-manager exits after 1min.
  process-managers.process-compose.settings.processes = {
    postgres.readiness_probe.exec.command = lib.mkForce "pg_isready -h 127.0.0.1 -p 5432 -d template1";
  };

  services.postgres = {
    enable = true;
    package = tpDeps.postgresql;
    extensions = extensions: [
      (import ./nix/prod/packages/pgvector.nix {
        inherit pkgs;
        postgresql = tpDeps.postgresql;
      })
    ];
    listen_addresses = "127.0.0.1";
    port = 5432;
  };
}
```

tp-app-deps.nix
```nix
{ pkgs }:

rec {
  elixir = pkgs.beam.packages.erlang_26.elixir_1_15;
  erlang = pkgs.erlang_26;
  postgresql = pkgs.postgresql_16;

  terrapin-app-deps = [
    pkgs.gnumake
    elixir
    pkgs.bash
    pkgs.nodejs
    pkgs.bws
    pkgs.qpdf
    pkgs.gnumeric # ssconvert
    pkgs.poppler_utils # pdf2text
  ];
}
```

devenv.yaml
```yaml
allowUnfree: true
allowBroken: true
inputs:
  nixpkgs:
    # If you update this, update nix/prod/flake.nix to ensure identical env between dev and prod
    url: github:nixos/nixpkgs/53488412621165b65bc27627fdb4057d65458d97
  agenix:
    url: github:ryantm/agenix/f6291c5935fdc4e0bef208cfc0dcab7e3f7a1c41
```

**Comment by pjar** [comment]
Same issue here. Looks like it might be related to https://github.com/cachix/devenv/issues/1394

**Comment by domenkozar** [comment]
@jozuas I've pushed a bit fix that should give us at least the version in the modules, can you `devenv update` and see what happens?

**Comment by domenkozar** [comment]
@jozuas can you run `devenv --refresh-eval-cache shell` and see if it remains?

**Comment by pjar** [comment]
@domenkozar For me the issue is gone after running `devenv --refresh-eval-cache shell`. Running the `devenv update` was not enough. Thank you for quick action on this.

**Comment by domenkozar** [comment]
This is fixed in 1.3.1, please upgrade once it reaches nixpkgs today!

**Comment by jozuas** [comment]
Upgrading to 1.3.1 fixes the problem
